_tools/libddwaf-updater/update.go

```diff
@@ -0,0 +1,284 @@
+			"// under the Apache License Version 2.0.",
+			"// This product includes software developed at Datadog (https://www.datadoghq.com/).",
+			"// Copyright 2016-present Datadog, Inc.",
+			"",
+			fmt.Sprintf("//go:build %s && %s && !go1.22", tgt.os, tgt.arch),
+			"package vendor",
+			"",
+			`import _ "embed" // Needed for go:embed`,
+			"",
+			fmt.Sprintf("//go:embed %s-%s/libddwaf.%s", tgt.os, tgt.arch, ext),
+			"var libddwaf []byte",
+			"",
+			fmt.Sprintf(`const embedNamePattern = "libddwaf-*.%s"`, ext),
+			"", // Trailing new line...
+		},
+		"\n",
+	)
+	if err := os.WriteFile(path.Join(libDir, fmt.Sprintf("vendor_%s_%s.go", tgt.os, tgt.arch)), []byte(gosource), 0644); err != nil {
+		panic(err)
+	}
+}
+
+func handleTarget(wg *sync.WaitGroup, version string, tgt target, embedDir string, assets map[string]*github.ReleaseAsset) {
+	defer wg.Done()
+
+	tarName := fmt.Sprintf("libddwaf-%s-%s.tar.gz", version, tgt.assetLabel)
+	shaName := fmt.Sprintf("%s.sha256", tarName)
+
+	tarAsset, found := assets[tarName]
+	if !found {
+		panic(fmt.Errorf("could not find tarball named %s", tarName))
+	}
+	shaAsset, found := assets[shaName]
+	if !found {
+		panic(fmt.Errorf("could not find sha256 named %s", shaName))
+	}
+
+	tarUrl := *tarAsset.BrowserDownloadURL
+	shaUrl := *shaAsset.BrowserDownloadURL
+
+	tmpdir, err := os.MkdirTemp("", "libddwaf-*")
+	if err != nil {
+		panic(err)
+	}
+	defer os.RemoveAll(tmpdir)
+
+	if _, err := script.Get(tarUrl).WriteFile(path.Join(tmpdir, tarName)); err != nil {
+		panic(err)
+	}
+	sha, err := script.Get(shaUrl).String()
+	if err != nil {
+		panic(err)
+	}
+	sum, err := script.File(path.Join(tmpdir, tarName)).SHA256Sum()
+	if err != nil {
+		panic(err)
+	}
+	// To match the shasum format...
+	sum = fmt.Sprintf("%s  %s\n", sum, tarName)
+	if sum != sha {
+		panic(fmt.Errorf("checksum mismatch on %s:\nExpected %s\nActual   %s", tarUrl, sha, sum))
+	}
+
+	file, err := os.Open(path.Join(tmpdir, tarName))
+	if err != nil {
+		panic(err)
+	}
+	reader, err := gzip.NewReader(file)
+	if err != nil {
+		panic(err)
+	}
+	arch := gotar.NewReader(reader)
+	foundLib := false
+	foundHdr := false
+	for {
+		header, err := arch.Next()
+		if err != nil {
+			if errors.Is(err, io.EOF) {
+				break
+			}
+			panic(err)
+		}
+
+		var destPath string
+		switch name := header.FileInfo().Name(); name {
+		case "libddwaf.so", "libddwaf.dylib":
+			destPath = path.Join(embedDir, name)
+			foundLib = true
+		case "ddwaf.h":
+			if tgt.primary {
+				destPath = path.Join(rootDir, "include", name)
+				foundHdr = true
+			} else {
+				continue
+			}
+		default:
+			continue
+		}
+
+		fmt.Printf("... downloaded %s\n", destPath)
+		if _, err := script.NewPipe().WithReader(arch).WriteFile(destPath); err != nil {
+			panic(err)
+		}
+		// Make the libraries executable, as this can be useful to link directly to those objects to perform troubleshooting.
+		if err := os.Chmod(destPath, 0755); err != nil {
+			panic(err)
+		}
```

Comment:
If I am right, the current code will also make the ddwaf.h file executable. Is it fine ?